lib/main.dart

```diff
@@ -24,10 +24,20 @@ class MyApp extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
+    final AppRoutes appRoutes = AppRoutes(repository);
+    final pageController = PageController();
+    final pageList = appRoutes
+        .getMovieEndpointRoutes(pageController)
```

Comment:
You can directly remove the pageController var and add it on line 30

Reply:
ahh, I see, that was a leftover from when I tried to access the page controller within the views, turns out I refactored that but forgot to take the pageController variable out of the view classes. Fixed now!